plugin/helpers_kv.go

```diff
@@ -129,3 +130,91 @@ func (p *HelpersImpl) KVSetWithExpiryJSON(key string, value interface{}, expireI
 
 	return nil
 }
+
+type kvListOptions struct {
+	KVListPerPage int
+	Prefix        string
```

Comment:
1/5 why not (also no need to special case "" I believe)
```go
func WithPrefix(prefix string) KVListOption {
	return WithChecker(f func(key string) (keep bool, err error) {
		return strings.HasPrefix(key, prefix), nil
	})
}
```

Reply:
The only notable difference is that Checker functions don't stack.  In the current implementation you can combine `WithPrefix` and `WithChecker` to get a first pass prefix filter with a filter function on the keys that remain.  I don't think that's important behavior based on the use cases, but I wanted to point it out in case anybody feels differently.

I'll remove the Prefix special case and update the comments to be more clear that it's either `WithPrefix` OR `WithChecker` but not both.

Reply:
Documenting the odd interplay between `WithPrefix` and `WithChecker` made it clear this was an implementation detail that wouldn't be clear at all to callers.  I've updated the implementation and made Checkers a slice of functions so they could stack and behave in a less surprising way.

Reply:
@levb I'm not convinced that the change has made the code more clear. Previously the internal structure mirrored the API, which made the code easy to understand. I'm not blocking on this, but still wanted to leave my thoughts.